functions/src/modules/mewm/events/updateEvent.ts

```diff
@@ -1,5 +1,83 @@
-import { ExpressFunction } from "../../../@types";
+import type { ExpressFunction } from "../../../@types";
+import { firebaseApp } from "../../../config/firebaseConfig";
+import {
+  expressErrorHandlerFactory,
+  requestBodyTypeValidator,
+} from "../../../helpers";
+import { uasPermissionSwitch } from "../../../systems/uas";
 
-const replaceMe: ExpressFunction = (req, res, next) => {};
+const updateEvent: ExpressFunction = (req, res, next) => {
+  uasPermissionSwitch({
+    organizer: { accepted: validateOrganizer },
+    sponsor: { accepted: validateSponsor }, // TO-DO: Need to filter only sponsor event managers allowed permission here
+  })(req, res, next);
+};
 
-export default replaceMe;
+const validateOrganizer: ExpressFunction = (req, res, next) => {
+  const validationRules: ValidatorObjectRules = {
+    type: "object",
+    rules: {
+        eventId: { rules: ["string"], required: true },
+        eventName: { rules: ["string"], required: true },
+        eventDescription: { rules: ["string"], required: true },
+        virtual: { rules: ["boolean"], required: true },
+        location: { rules: ["string"]},
+        joinLink: { rules: ["string"]},
+        joinLinkToPresenters: { rules: ["string"]},
+        joinLinkToAttendees: { rules: ["string"]},
+        companyId: { rules: ["string"] },
+        managers: { rules: [{
+          type: "array",
+          rules: ["string"],
+        }]},
+        speakers: { rules: [{
+          type: "array",
+          rules: ["string"],
+        }]}
+    },
+  };
+  requestBodyTypeValidator(req, res, next)(validationRules, execute);
+};
+
+const validateSponsor: ExpressFunction = (req, res, next) => {
+    const validationRules: ValidatorObjectRules = {
+      type: "object",
+      rules: {
+        eventId: { rules: ["string"], required: true },
+        eventName: { rules: ["string"], required: true },
+        eventDescription: { rules: ["string"], required: true },
+        virtual: { rules: ["boolean"], required: true },
+        location: { rules: ["string"]},
+        joinLink: { rules: ["string"]},
+        joinLinkToPresenters: { rules: ["string"]},
+        joinLinkToAttendees: { rules: ["string"]},
+        companyId: { rules: ["string"] },
+        managers: { rules: [{
+          type: "array",
+          rules: ["string"],
+        }]},
+        speakers: { rules: [{
+          type: "array",
+          rules: ["string"],
+        }]}
+      },
+    };
+    requestBodyTypeValidator(req, res, next)(validationRules, execute);
+  };
+
+const execute: ExpressFunction = (req, res, next) => {
+  const errorHandler = expressErrorHandlerFactory(req, res, next);
+
+  firebaseApp
+    .firestore()
+    .collection("miniEvents")
+    .doc(req.params.eventId)
+    .update(res.locals.parsedBody as METMMiniEventUpdateRequest)
```

Comment:
Heads up! It's the "Mini Events and Workshops Module" (MEWM). Also, I think that these are just referred to as "Events" and not "Mini Events" as they can also include things like deadlines or meals.
```suggestion
    .update(res.locals.parsedBody as MEWMEventUpdateRequest)
```